# No Implicit Buggy Globals

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```js

```

ğŸ‘ Examples of **correct** code for this rule:

```js

```

## When Not To Use It

TODO

## Version

4.3.2
