# Disallow `Element.prototype.innerText` in favor of `Element.prototype.textContent` (`github/no-innerText`)

ğŸ’¼ This rule is enabled in the ğŸ” `browser` config.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## Rule Details

ğŸ‘ Examples of **incorrect** code for this rule:

```js
const el = document.createElement('div')
el.innerText = 'foo'
```

ğŸ‘ Examples of **correct** code for this rule:

```js
const el = document.createElement('div')
el.textContent = 'foo'
```

## Version

4.3.2
